hw19/js/script.js

```diff
@@ -0,0 +1,232 @@
+  const arrayOfMarks = [];
+  while (arrayOfMarks.length < count) {
+    const randomMark =
+      Math.floor(Math.random() * (maxMark - minMark)) + minMark;
+    arrayOfMarks.push(randomMark);
+  }
+  return arrayOfMarks;
+}
+
+let student1 = new Student({
+  name: "Michael",
+  surname: "Brink",
+  age: 18,
+  marks: getArrayOfMarks(5, 1, 10),
+});
+let student2 = new Student({
+  name: "Mark",
+  surname: "Huber",
+  age: 20,
+  marks: getArrayOfMarks(5, 1, 10),
+});
+let student3 = new Student({
+  name: "Tom",
+  surname: "Vinston",
+  age: 25,
+  marks: getArrayOfMarks(5, 1, 10),
+});
+let student4 = new Student({
+  name: "Den",
+  surname: "Gruber",
+  age: 22,
+  marks: getArrayOfMarks(5, 1, 10),
+});
+let student5 = new Student({
+  name: "Jonh",
+  surname: "Brown",
+  age: 19,
+  marks: getArrayOfMarks(5, 1, 10),
+});
+
+console.log(student1);
+console.log("Michael's average mark " + student1.getAverageMark());
+console.log("Mark's average mark " + student2.getAverageMark());
+console.log("Tom's average mark " + student3.getAverageMark());
+console.log("Den's average mark " + student4.getAverageMark());
+console.log("Jonh's average mark " + student5.getAverageMark());
+console.log("Michael's min mark " + student1.getMinMark());
+console.log("Michael's max mark " + student1.getMaxMark());
+
+class FakeStudent extends Student {
+  #chetedMarks = [];
+  constructor({ name, surname, age, marks }) {
+    super({ name, surname, age, marks });
+    this.#cheat();
+  }
+
+  #cheat() {
+    for (let key of this.marks) {
+      const twicer = key * 2;
+      if (twicer > 10) {
+        this.#chetedMarks.push(10);
+      } else {
+        this.#chetedMarks.push(twicer);
+      }
+    }
+    return this.#chetedMarks;
+  }
+
+  getCheat() {
+    return this.#chetedMarks;
+  }
+
+  getAverageMark() {
+    let summaOfMarks = 0;
+    this.#chetedMarks.forEach((item) => {
+      summaOfMarks += item;
+    });
+    return summaOfMarks / this.#chetedMarks.length;
+  }
+
+  getMinMark() {
+    return Math.min(...this.#chetedMarks);
+  }
+
+  getMaxMark() {
+    return Math.max(...this.#chetedMarks);
+  }
+}
+
+let fakeStudent = new FakeStudent({
+  name: "Leonardo",
+  surname: "DiCaprio",
+  age: 55,
+  marks: getArrayOfMarks(5, 1, 10),
+});
+let fakeStudent2 = new FakeStudent({
+  name: "Ledi",
+  surname: "Di",
+  age: 30,
+  marks: getArrayOfMarks(5, 1, 10),
+});
+
+console.log(fakeStudent);
+console.log("Array of cheted marks", fakeStudent.getCheat());
+console.log("Leonardo's average mark " + fakeStudent.getAverageMark());
+console.log("Leonardo's min mark " + fakeStudent.getMinMark());
+console.log("Leonardo's max mark " + fakeStudent.getMaxMark());
+
+class Teacher extends Student {
+  constructor({ name, surname, age, group }) {
+    super({ name, surname, age });
+    this.group = group;
+  }
+
+  getListOfNamesByAverageMark() {
+    this.group.sort((a, b) => b.getAverageMark() - a.getAverageMark());
```

Comment:
sort method mutates original value this.group (bad practice). So you need to use copy of this.group instead of original 
```javascript
const sortedGroup = [...this.group].sort((a, b) => b.getAverageMark() - a.getAverageMark());
```